src/v2/modules/invoice-accounts/validator.js

```diff
@@ -6,15 +6,28 @@ const Joi = require('@hapi/joi')
 const invoiceAccountNumberRegex = /^[ABENSTWY][0-9]{8}A$/;
 const DATE = Joi.date().utc().format('YYYY-MM-DD');
 
-const schema = Joi.object({
+const invoiceAccountSchema = Joi.object({
   companyId: Joi.string().guid().required(),
   invoiceAccountNumber: Joi.string().regex(invoiceAccountNumberRegex).required(),
   startDate: DATE.required(),
   endDate: DATE.min(Joi.ref('startDate')).optional().default(null).allow(null),
   isTest: Joi.boolean().optional().default(false)
 });
 
+const invoiceAccountAddressSchema = Joi.object({
+  invoiceAccountId: Joi.string().guid().required(),
+  addressId: Joi.string().guid().required(),
+  startDate: DATE.required(),
+  endDate: DATE.min(Joi.ref('startDate')).optional().default(null).allow(null),
+  isTest: Joi.boolean().optional().default(false)
+});
+
+const schemas = {
+  invoiceAccount: invoiceAccountSchema,
+  invoiceAccountAddress: invoiceAccountAddressSchema
+};
+
 /**
  * Validates that an object conforms to the requirements of a contact.
  */
-exports.validate = invoiceAccount => Joi.validate(invoiceAccount, schema, { abortEarly: false });
+exports.validate = (obj, key) => Joi.validate(obj, schemas[key], { abortEarly: false });
```

Comment:
Could this be replaced with

```
exports.validateInvoiceAccount = invoiceAccount => 
  Joi.validate(invoiceAccount, invoiceAccountSchema, { abortEarly: false });


exports.validateInvoiceAccountAddress = invoiceAccountAddress => 
  Joi.validate(invoiceAccountAddress, invoiceAccountAddressSchema, { abortEarly: false });
```

To me it feels like the consumer of this function has to know how it operates in order to use it.

Reply:
I considered making that change this morning, next time, I'll just go for it